server/src/operators/chunk_operator.rs

```diff
@@ -1531,16 +1563,38 @@ pub async fn create_chunk_metadata(
             }
         }
 
-        let group_ids_from_group_tracking_ids = if let Some(group_tracking_ids) =
-            chunk.group_tracking_ids.clone()
-        {
-            get_group_ids_from_tracking_ids_query(group_tracking_ids, dataset_uuid, pool.clone())
-                .await
-                .ok()
-                .unwrap_or(vec![])
-        } else {
-            vec![]
-        };
+        let group_ids_from_group_tracking_ids =
+            if let Some(group_tracking_ids) = chunk.group_tracking_ids.clone() {
+                let group_id_tracking_ids = get_group_ids_from_tracking_ids_query(
+                    group_tracking_ids.clone(),
+                    dataset_uuid,
+                    pool.clone(),
+                )
+                .await?;
+
+                let group_ids = group_id_tracking_ids
+                    .clone()
+                    .into_iter()
+                    .map(|(group_id, _)| group_id)
+                    .collect::<Vec<uuid::Uuid>>();
+                let found_group_tracking_ids = group_id_tracking_ids
+                    .into_iter()
+                    .filter_map(|(_, group_tracking_id)| group_tracking_id)
+                    .collect::<Vec<String>>();
+
+                for group_tracking_id in group_tracking_ids {
+                    if !found_group_tracking_ids.contains(&group_tracking_id) {
+                        return Err(ServiceError::BadRequest(format!(
+                            "Group with tracking id {} does not exist",
+                            group_tracking_id
+                        )));
+                    }
+                }
```

Comment:
```suggestion
```
  whats the context behind this? If it stops at this point the full transaction will be reverted. We already check if the group_tracking_id exists in the handler here.    https://github.com/devflowinc/trieve/blob/adb27fea33e89553b6d2cb02ec7e6d86b1c54414/server/src/operators/chunk_operator.rs#L1547-L1597
  
 If it gets to this point and it doesn't exist, then they deleted it mid ingest which means we either ingested too slowly. Or a freak race condition occurred

Reply:
Confusing function name, but this actually gets called from the handler. We already had a check for `group_id` but not `group_tracking_id`.